src/codemagic/models/certificate.py

```diff
@@ -101,8 +101,15 @@ def is_development_certificate(self) -> bool:
         return development_certificate_pattern.match(self.common_name) is not None
 
     def is_code_signing_certificate(self) -> bool:
-        code_signing_certificate_pattern = re.compile(
-            r'^((Apple (Development|Distribution))|(iPhone (Developer|Distribution))):.*$')
+        code_signing_certificate_pattern = re.compile((
```

Comment:
I find that this regex has become too complex and it would benefit if we chop it down to pieces and operate on those smaller chunks instead. I suggest the followind refactoring:
```python
    def is_code_signing_certificate(self) -> bool:
        code_signing_certificate_patterns = (
            re.compile(r'^Apple (Development|Distribution):.*$'),
            re.compile(r'^iPhone (Developer|Distribution):.*$'),
            re.compile(r'^Developer ID Application:.*$'),
            re.compile(r'^Mac Developer:.*$'),
            re.compile(r'^3rd Party Mac Developer (Application|Installer):.*$'),
        )
        return any(p.match(self.common_name) is not None for p in code_signing_certificate_patterns)
```
and also add this test case to `tests/models/test_certificate.py`:
```python
@pytest.mark.parametrize('cert_common_name, is_code_signing_certificate', [
    ('Apple Development: Some name (FXZPHT7PIC)', True),
    ('Apple Distribution: Some name (FXZPHT7PIC)', True),
    ('iPhone Developer: Some name (FXZPHT7PIC)', True),
    ('iPhone Distribution: Some name (FXZPHT7PIC)', True),
    ('Developer ID Application: Some name (FXZPHT7PIC)', True),
    ('Mac Developer: Some name (FXZPHT7PIC)', True),
    ('3rd Party Mac Developer Application: Some name (FXZPHT7PIC)', True),
    ('3rd Party Mac Developer Installer: Some name (FXZPHT7PIC)', True),
    ('iPhone Development: Some name (FXZPHT7PIC)', False),
    ('Apple Worldwide Developer Relations Certification Authority', False),
])
def test_is_code_signing_certificate(cert_common_name, is_code_signing_certificate, certificate_asn1):
    patched_common_name = mock.PropertyMock(return_value=cert_common_name)
    with mock.patch.object(Certificate, 'common_name', new_callable=patched_common_name):
        certificate = Certificate.from_ans1(certificate_asn1)
        assert certificate.is_code_signing_certificate() is is_code_signing_certificate
```

Reply:
fixed in d7de32a87dd4bfd264c78c390a182d62a95526cd